### DISCUSSION TITLE
Folder empty after installation

### DISCUSSION THREAD

**Discussion by zotori** [jamulussoftware/jamulus#2596] [Technical Support & Help]
Hi, I'm a newbie at both jamulus and linux. 

I've opened up an Ubuntu instance in Ali's Cloud.

- As the Jamulus Wiki suggested, I'v downloaded the Jamulus .deb file (https://jamulus.io/wiki/Installation-for-Linux) 
- I've successfully moved the file to my linux server, ran the .deb file using command, (sudo apt install ./jamulus... .deb )
- the result gave me an error message :
   "N: Download is performed unsandboxed as root as file '/root/jamulusexe.deb' couldn't be accessed by user '_apt'. - pkgAcquire::Run (13: Permission denied)"
- through search, I realized this could be the issue with writing permission, and I solved issue by the following commands
   (sudo chown -Rv _apt:root /var/cache/apt/archives/partial/)
   (sudo chmod -Rv 700 /var/cache/apt/archives/partial/)
- I ran the installer again, (sudo apt install ./jamulus... .deb ) and I got the following result message 
   
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Note, selecting 'jamulus' instead of './jamulusexe.deb'
jamulus is already the newest version (3.8.2-1).
0 upgraded, 0 newly installed, 0 to remove and 7 not upgraded.

- so at this point, I believe the installation is done, however, when I type in command 'ls', I don't see any related file in the list, its just empty, nor I can launch QjackCtl, which is the next suggested step according to Jamulus Wiki.

Please help! thank you very much.

**Comment by ann0see** [comment]
Strange. The deb works fine for me.
By running ls you won't necessarily see the files. You're basically just seeing what is in your current folder.


Did you install qjackctl? What happens if you try to open qjackctl via command line?
What are the exact commands you entered?
What happens if you run Jamulus via typing `jamulus` in the terminal?

**Reply by zotori** [reply]
Thanks for helping!

- I didn't install QjackCtl separatly
- I assume ver

But when typed in jamulus in the terminal, 
I get the following message : 

- allocated port number: 22134
qt.qpa.xcb: could not connect to display 
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.


Maybe I should try to erase everything and re-install?

**Reply by hoffie** [reply]
I think even your first installation attempt looked fine. The warning you've got was probably related to the fact that you've placed the downloaded file in a directory which was unaccessible to apt's unprivileged mode. That's no issue for local downloads though and apt falls back to just doing it with a privileged user.

You've mentioned "Ali's cloud" so I assume you are planning to run a server without a GUI, right?

Then the *headless* download would have been all you've needed. Now that you've got the "full" version, you can also use this.

By default, Jamulus starts as a client with a full GUI. Based on your logs, this fails -- most likely because you are not running a GUI at all (?).

So, if you want to run a non-GUI Jamulus server, you'd want `jamulus --server --nogui`.
The following docs provide some further details:

https://jamulus.io/de/wiki/Server-Linux

**Reply by ann0see** [reply]
Ah yes. If you want to run a server, you should install the headless version – or even easier – use a pre configured jamulus provider. 

Just a personal warning:
Running a Linux server in the cloud needs some experience, can be time consuming and needs some basic security measures to ensure your server doesn't get mis-used by others ("=ensure it's not easily hackable")

At least make sure your server uses a strong password and is protected such that you don't have bad experiences security wise. I suggest to read a bit about how to secure a Linux server before running one. (You can Google for some guides on securing a Linux server).

Maybe we should also put this disclaimer on the headless server page
